astropy/convolution/convolve.py

```diff
@@ -869,3 +887,30 @@ def convolve_models(model, kernel, mode='convolve_fft', **kwargs):
         raise ValueError(f'Mode {mode} is not supported.')
 
     return CompoundModel(mode, model, kernel)
+
+
+def _next_fast_lengths(shape):
+    """
+    Find optimal or good sizes to pad an array of ``shape`` to for better
+    performance with `numpy.fft.*fft` and `scipy.fft.*fft`.
+    Calculated directly with `scipy.fft.next_fast_len`, if available; otherwise
+    looked up from list and scaled by powers of 10, if necessary.
+    """
+
+    try:
+        import scipy.fft
+        return np.array([scipy.fft.next_fast_len(j, real=False) for j in shape])
+    except ImportError:
+        pass
+
+    newshape = np.empty(len(np.atleast_1d(shape)), dtype=int)
+    for i, j in enumerate(shape):
+        scale = 10 ** max(int(np.ceil(np.log10(j))) - _good_range, 0)
+        for n in _good_sizes:
+            if n * scale >= j:
+                newshape[i] = n * scale
```

Comment:
could this be more efficiently implemented as:
```python
scales = _good_sizes * scale >= j
if np.any(scales):
    newshape[i] = np.argmax(scales)
else: 
    raise
```

It seems like a small thing, but I always try to avoid nested loops.  In this case, it's _probably_ negligible though.

Reply:
Thanks for the review; I just started being unhappy with the Python loop as well, but used `searchsorted` now to make it a single statement.